src/v/cluster/partition.h

```diff
@@ -280,6 +280,13 @@ class partition {
 
     uint64_t upload_backlog_size() const;
 
+    /**
+     * Partition 0 carries a copy of the topic configuration, updated by
+     * the controller, so that its archiver can make updates to the topic
+     * manifest in cloud storage
+     */
+    void set_topic_config(std::unique_ptr<cluster::topic_configuration> cfg);
```

Comment:
nit: maybe `set_archiver_topic_config` would be a more suitable name? Makes the scope of the config update clear.

Reply:
Yeah, here it's kind of gross either way: make the interface conspicuously leaky by calling out in the name that this is really only here for one of the sub objects' benefit, or make it vaguely obscure.  I think I'll leave this, and plan on stripping out this whole weird "partition 0 gets the topic config" when we do general metadata uploads.